### PR TITLE
feat(sqlexec rpc): client auth error messages

### PR DESCRIPTION
**Pull request by greyscaled**
Examples:

1. Auth errors (connection string, `--cloud-addr` mismatch etc)
```shell
$ cargo run --bin glaredb -- --cloud-url=<URL> --cloud-addr=https://console.glaredb.com
GlareDB (v0.6.1)
Error: authentication failed
```

2. Version out of date (0.7.0):

```shell
$ cargo run --bin glaredb -- --cloud-url=<URL> --cloud-addr=https://qa.glaredb.com
GlareDB (v0.6.1)
Error: version unsupported: update to glaredb v0.7.0 or later: hint: curl https://glaredb.com/install.sh | sh
```

### CODE REVIEW

File: crates/sqlexec/src/remote/client.rs [hunk 1]
```diff
@@ -209,26 +214,32 @@ impl RemoteClient {
             body.insert("password", params.password);
             body.insert("org_name", params.org);
             body.insert("db_name", params.db_name);
-            body.insert("api_version", 2.to_string());
+            body.insert("api_version", 1.to_string());
```
Comment (greyscaled) [inline crates/sqlexec/src/remote/client.rs:217]: Merge: Change back

File: crates/sqlexec/src/remote/client.rs [hunk 2]
```diff
@@ -199,35 +204,44 @@ impl RemoteClient {
         metadata.insert(DB_NAME_KEY, params.db_name.parse()?);
         metadata.insert(ORG_KEY, params.org.parse()?);
 
+        let mut body = HashMap::new();
+        body.insert("user", params.user);
+        body.insert("password", params.password);
+        body.insert("org_name", params.org);
+        body.insert("db_name", params.db_name);
+        body.insert("api_version", 1.to_string());
```
Comment (greyscaled) [inline crates/sqlexec/src/remote/client.rs:212]:
```suggestion
        body.insert("api_version", 2.to_string());
```

File: crates/sqlexec/src/remote/client.rs [hunk 3]
```diff
@@ -199,35 +204,44 @@ impl RemoteClient {
...
 
+        let mut body = HashMap::new();
+        body.insert("user", params.user);
+        body.insert("password", params.password);
+        body.insert("org_name", params.org);
+        body.insert("db_name", params.db_name);
+        body.insert("api_version", 2.to_string());
+        body.insert("client_type", client_type.to_string());
+
+        info!("client authentication");
```
Comment (tychoish) [inline crates/sqlexec/src/remote/client.rs:215]: debug?

File: crates/sqlexec/src/remote/client.rs [hunk 4]
```diff
@@ -199,35 +204,44 @@ impl RemoteClient {
...
+            .json(&body)
+            .send()
+            .await?;
+
+        if res.status() != reqwest::StatusCode::OK {
+            if res.status().is_client_error() {
+                let err = res.json::<AuthenticateClientError>().await?;
+                return Err(ExecError::String(err.msg));
+            } else {
+                return Err(ExecError::Internal("client authentication".to_string()));
```
Comment (tychoish) [inline crates/sqlexec/src/remote/client.rs:231]: should we put the error code here?

File: crates/sqlexec/src/remote/client.rs [hunk 5]
```diff
@@ -199,35 +204,44 @@ impl RemoteClient {
...
-                .post(format!(
-                    "{}/api/internal/authenticate/client",
-                    cloud_api_addr
-                ))
-                .json(&body)
-                .send()
-                .await?
-                .json::<AuthenticateClientResponse>()
-                .await?;
+            info!("apply TLS certificate");
```
Comment (tychoish) [inline crates/sqlexec/src/remote/client.rs:238]: debug

File: crates/sqlexec/src/remote/client.rs (hunk 3 above)
Comment (greyscaled) [reply crates/sqlexec/src/remote/client.rs:215]: I had it here intentionally actually, but lmk if you think it shouldn't be

File: crates/sqlexec/src/remote/client.rs (hunk 5 above)
Comment (greyscaled) [reply crates/sqlexec/src/remote/client.rs:238]: This was there before, and I don't mind keeping it in for when verbose logs are on.

File: crates/sqlexec/src/remote/client.rs (hunk 4 above)
Comment (greyscaled) [reply crates/sqlexec/src/remote/client.rs:231]: That sgtm

**Review by scsmithr — APPROVED** [review]
w/ debug

**Review by tychoish — APPROVED** [review]

**Pull request merged**